src/ai/backend/common/validators.py

```diff
@@ -696,3 +698,26 @@ def check_and_return(self, value: Any) -> float:
                 return 0
             case _:
                 self._failure(f"Value must be (float, tuple of float or None), not {type(value)}.")
+
+
+class RoundRobinStatesJSONString(t.Trafaret):
+    def check_and_return(self, value: Any) -> RoundRobinStates:
+        if not isinstance(value, str):
+            self._failure("RoundRobinStatesJSONString must be JSON string", value=value)
+        else:
+            try:
+                rr_states_dict: dict[str, dict[str, Any]] = json.loads(value)
+            except (KeyError, ValueError):
+                self._failure("RoundRobinStatesJSONString is not a valid JSON string", value=value)
+
+            rr_states = {}
+            for arch, rr_state_dict in rr_states_dict.items():
+                if "next_index" not in rr_state_dict or "schedulable_group_id" not in rr_state_dict:
+                    self._failure("Invalid roundrobin states")
+
+                rr_states[arch] = RoundRobinState(
+                    schedulable_group_id=cast(str, rr_state_dict.get("schedulable_group_id")),
+                    next_index=cast(int, rr_state_dict.get("next_index")),
+                )
```

Comment:
I have one idea about this part.
if `RoundRobinState` inherits `JSONSerializableMixin`, then `RoundRobinState` can validate by itself.
```python
for arch, rr_state_dict in rr_states_dict.items():
    rr_states[arch] = RoundRobinState.from_json(rr_state_dict)
```
How do you think of it?